### ISSUE TITLE
RuntimeError(emojis(f"Dataset '{clean_url(self.args.data)}' error ❌ {e}")) from e

### ISSUE THREAD

**Issue by arnobjohn** [ultralytics/ultralytics#14324]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I am trying to run the yolov8 model on a custom dataset and it outputs the following
![image](https://github.com/ultralytics/ultralytics/assets/58500422/40a2e3e0-224f-4949-946b-5e3fcfe16658)

My project is arranged in the following manner:
![image](https://github.com/ultralytics/ultralytics/assets/58500422/bdc6fc51-349d-44ce-8185-53f903d42cf0)

train.py file content:
![image](https://github.com/ultralytics/ultralytics/assets/58500422/cb876faa-861d-4d9b-9b78-9394ba18405c)

and, data.yaml file content:
![image](https://github.com/ultralytics/ultralytics/assets/58500422/f3f5bae8-4160-4f56-9ecf-7c463026a68f)

I am trying to figure out where is the issue.

### Additional

A similar question was asked before.

**Comment by glenn-jocher** [comment]
@arnobjohn hi there! 👋

Thank you for providing detailed information about your issue. It looks like you're encountering a `RuntimeError` related to your dataset configuration. Here are a few steps to help you troubleshoot and resolve this:

1. **Verify Dataset Paths**: Ensure that the paths specified in your `data.yaml` file are correct and accessible. The paths should point to the directories containing your training and validation images and labels.

2. **Check Dataset Structure**: Your dataset directories should follow the expected structure. For example:
   ```
   dataset/
   ├── images/
   │   ├── train/
   │   └── val/
   └── labels/
       ├── train/
       └── val/
   ```

3. **File Extensions**: Ensure that the image and label files have the correct extensions and match the entries in your `data.yaml` file.

4. **Permissions**: Verify that your script has the necessary permissions to read the dataset files.

5. **Reproducible Example**: If the issue persists, please provide a minimum reproducible example to help us diagnose the problem more effectively. You can find guidelines on how to create one here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

6. **Update Packages**: Ensure you are using the latest versions of the Ultralytics and YOLOv8 packages. Sometimes, issues are resolved in newer releases.

Here's a quick checklist to verify your `data.yaml` file:
```yaml
train: /path/to/your/dataset/images/train
val: /path/to/your/dataset/images/val

nc: 80  # number of classes
names: ['class1', 'class2', ...]  # list of class names
```

If you continue to face issues, please share any additional error messages or logs that might help us pinpoint the problem. We're here to help! 😊

**Comment by arnobjohn** [comment]
@glenn-jocher Hi

I have tried changing the dataset structure:
![image](https://github.com/ultralytics/ultralytics/assets/58500422/37009c26-8016-41de-bd2a-54e3a8d9aa39)

even after that there seems to be no change in output:
![image](https://github.com/ultralytics/ultralytics/assets/58500422/01f2603a-41cf-4ac4-befa-8c4f145204c3)

changes done to data.yaml:
![image](https://github.com/ultralytics/ultralytics/assets/58500422/c9e1345d-8386-4379-8ab8-da1f47a965d7)

Should I be changing the location of data.yaml folder to the location wheree train.py is present?:
![image](https://github.com/ultralytics/ultralytics/assets/58500422/e73b3919-549b-418e-b7cf-2d31b749241a)


Environment: I am running the code on a remote python server.

**Comment by glenn-jocher** [comment]
Hi @arnobjohn,

Thank you for the update and for providing additional details. Let's try to resolve this issue together! 😊

### Steps to Troubleshoot:

1. **Verify `data.yaml` Paths**:
   Ensure that the paths in your `data.yaml` file are absolute or correctly relative to the location where you are running your script. For example:
   ```yaml
   train: /absolute/path/to/dataset/images/train
   val: /absolute/path/to/dataset/images/val

   nc: 80  # number of classes
   names: ['class1', 'class2', ...]  # list of class names
   ```

2. **Check Dataset Structure**:
   Your dataset structure should look like this:
   ```
   dataset/
   ├── images/
   │   ├── train/
   │   └── val/
   └── labels/
       ├── train/
       └── val/
   ```

3. **Location of `data.yaml`**:
   The `data.yaml` file does not need to be in the same directory as `train.py`, but it should be accessible from the script. Ensure the path you provide in your script points correctly to `data.yaml`.

4. **Permissions**:
   Verify that your script has the necessary permissions to read the dataset files.

5. **Reproducible Example**:
   If the issue persists, please provide a minimum reproducible example. This helps us diagnose the problem more effectively. You can find guidelines here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

6. **Update Packages**:
   Ensure you are using the latest versions of the Ultralytics and YOLOv8 packages. Sometimes, issues are resolved in newer releases.

### Example Code:
Here’s a quick example to ensure your paths are correctly set:
```python
from ultralytics import YOLO

# Load the model
model = YOLO('yolov8n.pt')

# Train the model
model.train(data='/absolute/path/to/data.yaml', epochs=100, imgsz=640)
```

### Additional Tips:
- **Check Logs**: Look at the logs for any specific error messages that might give more insight into what’s going wrong.
- **Environment**: Ensure your remote server has all the necessary dependencies installed and is configured correctly.

If you continue to face issues, please share any additional error messages or logs that might help us pinpoint the problem. We're here to help! 😊

**Comment by arnobjohn** [comment]
I am facing this issue now:
![image](https://github.com/ultralytics/ultralytics/assets/58500422/a4f6c908-a904-45e5-8e60-9aa8859406d1)

where am I doing wrong?

**Comment by glenn-jocher** [comment]
Hi @arnobjohn,

Thank you for sharing the screenshot and additional details. Let's work through this together! 😊

### Steps to Troubleshoot:

1. **Verify Dataset Paths**:
   Ensure that the paths in your `data.yaml` file are absolute or correctly relative to the location where you are running your script. For example:
   ```yaml
   train: /absolute/path/to/dataset/images/train
   val: /absolute/path/to/dataset/images/val

   nc: 80  # number of classes
   names: ['class1', 'class2', ...]  # list of class names
   ```

2. **Check Dataset Structure**:
   Your dataset structure should look like this:
   ```
   dataset/
   ├── images/
   │   ├── train/
   │   └── val/
   └── labels/
       ├── train/
       └── val/
   ```

3. **Location of `data.yaml`**:
   The `data.yaml` file does not need to be in the same directory as `train.py`, but it should be accessible from the script. Ensure the path you provide in your script points correctly to `data.yaml`.

4. **Permissions**:
   Verify that your script has the necessary permissions to read the dataset files.

5. **Reproducible Example**:
   If the issue persists, please provide a minimum reproducible example. This helps us diagnose the problem more effectively. You can find guidelines here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

6. **Update Packages**:
   Ensure you are using the latest versions of the Ultralytics and YOLOv8 packages. Sometimes, issues are resolved in newer releases.

### Example Code:
Here’s a quick example to ensure your paths are correctly set:
```python
from ultralytics import YOLO

# Load the model
model = YOLO('yolov8n.pt')

# Train the model
model.train(data='/absolute/path/to/data.yaml', epochs=100, imgsz=640)
```

### Additional Tips:
- **Check Logs**: Look at the logs for any specific error messages that might give more insight into what’s going wrong.
- **Environment**: Ensure your remote server has all the necessary dependencies installed and is configured correctly.

If you continue to face issues, please share any additional error messages or logs that might help us pinpoint the problem. We're here to help! 😊

**Comment by arnobjohn** [comment]
@glenn-jocher Thank you for your valuable advice. 

Issue was with using incorrect absolute path. As I was using remote server, I just used the absolute path of the dataset uploaded in the remote server.

**Comment by glenn-jocher** [comment]
You're welcome! I'm glad to hear you identified the issue with the absolute path. Ensuring the correct paths, especially on remote servers, is crucial. If you encounter any further issues or have additional questions, feel free to ask.
